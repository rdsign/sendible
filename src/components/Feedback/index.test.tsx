import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Feedback } from './index';
import { Template } from '../../../spec/template';

describe('Feedback', () => {
    const component = (
        <Template>
            <Feedback messages={['message one', 'message two']} />
        </Template>
    );

    test('should match snapshot', () => {
        const snapshot = create(component);
        expect(snapshot).toMatchSnapshot();
    });

    test('feedback should cointain "• message one" and "• message two" inner text', () => {
        render(component);

        expect(screen.getByText('• message one')).toBeInTheDocument();
        expect(screen.getByText('• message two')).toBeInTheDocument();
    });
});
