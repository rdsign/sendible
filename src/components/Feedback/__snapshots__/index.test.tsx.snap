// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should match snapshot 1`] = `
.c0 {
  background: #ff6153;
  border-radius: 0.8rem;
  color: #ffffff;
  font-size: 1.6rem;
  left: 50%;
  line-height: 1.66em;
  padding: 2rem 3rem;
  position: fixed;
  top: -30rem;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
  visibility: hidden;
  z-index: 100;
  -webkit-animation: appear 8s ease-in-out;
  animation: appear 8s ease-in-out;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  visibility: visible;
}

<div
  className="c0"
  data-testid="feedback"
>
  <p>
    • 
    message one
  </p>
  <p>
    • 
    message two
  </p>
</div>
`;
